ffmpeg | branch: master | Carl Eugen Hoyos <cehoyos at ag.or.at> | Tue Mar 22 22:38:44 2011 +0100| [78e99e0f32b610a3cf302b2b35e500f4e9eaff07] | committer: Ronald S. Bultje

Do not use format string "%0.3f" for RTSP Range field.

The format string was locale-depending.

Signed-off-by: Ronald S. Bultje <rsbultje at gmail.com>

> http://git.videolan.org/gitweb.cgi/ffmpeg.git/?a=commit;h=78e99e0f32b610a3cf302b2b35e500f4e9eaff07
---

 libavformat/rtspdec.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

ffmpeg | branch: master | Martin Storsjö <martin at martin.st> | Wed Mar 23 10:26:22 2011 +0200| [28e9c42afbf0088aa1f4280a95d304dca4b675a0] | committer: Ronald S. Bultje

rtsp: Don't use a locale dependent format string

In this particular case, we aren't ever printing anything else than
0.000 anyway.

Signed-off-by: Ronald S. Bultje <rsbultje at gmail.com>

> http://git.videolan.org/gitweb.cgi/ffmpeg.git/?a=commit;h=28e9c42afbf0088aa1f4280a95d304dca4b675a0
---

 libavformat/rtspenc.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

 
diff --git a/lib/ffmpeg/libavformat/rtspdec.c b/lib/ffmpeg/libavformat/rtspdec.c
index 4f926e4..f619256 100644
--- a/lib/ffmpeg/libavformat/rtspdec.c
+++ b/lib/ffmpeg/libavformat/rtspdec.c
@@ -60,8 +60,9 @@ static int rtsp_read_play(AVFormatContext *s)
             cmd[0] = 0;
         } else {
             snprintf(cmd, sizeof(cmd),
-                     "Range: npt=%0.3f-\r\n",
-                     (double)rt->seek_timestamp / AV_TIME_BASE);
+                     "Range: npt=%"PRId64".%03"PRId64"-\r\n",
+                     rt->seek_timestamp / AV_TIME_BASE,
+                     rt->seek_timestamp / (AV_TIME_BASE / 1000) % 1000);
         }
         ff_rtsp_send_cmd(s, "PLAY", rt->control_uri, cmd, reply, NULL);
         if (reply->status_code != RTSP_STATUS_OK) {
diff --git a/lib/ffmpeg/libavformat/rtspenc.c b/lib/ffmpeg/libavformat/rtspenc.c
index 34deeeb..5d76fdf 100644
--- a/lib/ffmpeg/libavformat/rtspenc.c
+++ b/lib/ffmpeg/libavformat/rtspenc.c
@@ -103,8 +103,7 @@ static int rtsp_write_record(AVFormatContext *s)
     char cmd[1024];
 
     snprintf(cmd, sizeof(cmd),
-             "Range: npt=%0.3f-\r\n",
-             (double) 0);
+             "Range: npt=0.000-\r\n");
     ff_rtsp_send_cmd(s, "RECORD", rt->control_uri, cmd, reply, NULL);
     if (reply->status_code != RTSP_STATUS_OK)
         return -1;
